R/calc_depth.R

```diff
@@ -0,0 +1,41 @@
+#' Estimate depth from discharge and hydraulic geometry coefficients
+#' 
+#' Uses the relationship \eqn{d=c*Q^f} (parameter names and definitions as in 
+#' Leopold and Maddock, 1953; default values for c and f as in Raymond et al.
+#' 2012)
+#' 
+#' @param Q discharge (m^3 s^-1)
+#' @param c coefficient representing depth at unit discharge (usually m)
+#' @param f exponent in depth-discharge relation (unitless)
+#' @return d, stream depth, in the same units as c
+#' @examples
+#' calc_depth(Q=1:10)
+#' calc_depth(Q=1:10, f=0.4)
+#' library(unitted)
+#' calc_depth(Q=u(1:10, "m^3 s^-1"), c=u(40,"cm"))
+#' calc_depth(Q=u(1:10, "m^3 s^-1"), f=u(0.36))
+#' @importFrom unitted u v verify_units
+#' @export
+calc_depth <- function(Q, c=u(0.409,"m"), f=u(0.294,"")) {
+
+  with.units <- is.unitted(Q) || (if(!missing(c)) is.unitted(c) else FALSE) || (if(!missing(f)) is.unitted(f) else FALSE)
```

Comment:
I think I get at this with the test line that runs `calc_depth(Q=7.3)`, which has unitted defaults but lets them pass because with.units is set to FALSE because missing(c) and missing(f) are TRUE